src/main/java/com/exemple/supermarket/exception/RestTemplateResponseErrorHandler.java

```diff
@@ -0,0 +1,41 @@
+package com.exemple.supermarket.exception;
+
+import lombok.SneakyThrows;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.web.client.ResponseErrorHandler;
+
+import java.io.IOException;
+
+import static org.springframework.http.HttpStatus.Series.CLIENT_ERROR;
+import static org.springframework.http.HttpStatus.Series.SERVER_ERROR;
+
+public class RestTemplateResponseErrorHandler
+        implements ResponseErrorHandler {
+
+    @Override
+    public boolean hasError(ClientHttpResponse httpResponse)
+            throws IOException {
+
+        return (
+                httpResponse.getStatusCode().series() == CLIENT_ERROR
+                        || httpResponse.getStatusCode().series() == SERVER_ERROR);
+    }
+
+    @SneakyThrows
+    @Override
+    public void handleError(ClientHttpResponse httpResponse)
+            throws IOException {
+
+        if (httpResponse.getStatusCode()
+                .series() == HttpStatus.Series.SERVER_ERROR) {
+            // handle SERVER_ERROR
```

Comment:
so what do you think about handling this?

Reply:
done